tiledb/sm/array/array.cc

```diff
@@ -218,15 +219,12 @@ Status Array::evolve_array_schema(
 
   // Evolve schema
   auto array_schema_evolved = schema_evolution->evolve_schema(array_schema);
-
-  Status st =
-      store_array_schema(resources, array_schema_evolved, encryption_key);
-  if (!st.ok()) {
-    throw ArrayException(
-        "Cannot evolve schema;  Not able to store evolved array schema.");
+  try {
+    store_array_schema(resources, array_schema_evolved, encryption_key);
+  } catch (std::exception& e) {
```

Comment:
If the exception being caught is not referenced, there's no need to name it. This may generate a warning, one similar to "unused argument", in future compilers.